### PR TITLE
Updated recast js

### PR DESCRIPTION
**Pull request by CedricGuillemet**
No new functionalities. Only recastjs built with latest emscripten and recastnavigation.
Tested with every PG in documentation with and without worker thread.

incoming PR on extensions.

### CODE REVIEW

File: src/Navigation/Plugins/recastJSPlugin.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export class RecastJSPlugin implements INavigationEnginePlugin {
      */
     public constructor(recastInjection: any = Recast) {
         if (typeof recastInjection === "function") {
-            recastInjection(this.bjsRECAST);
```
Comment (RaananW) [inline src/Navigation/Plugins/recastJSPlugin.ts:47]: is `await Recast()` generate anything in the global namespace? no way to inject an import anymore?

Comment (CedricGuillemet) [reply src/Navigation/Plugins/recastJSPlugin.ts:47]: I'm not the best to answer these questions :) I build recastjs the same way as ammojs and load it the same way as well.

Comment (RaananW) [reply src/Navigation/Plugins/recastJSPlugin.ts:47]: Oh, just saw the ammo init, which has the same log. I assume that you can await this function instead, but i'm not not entirely sure without checking.

**Review by RaananW — APPROVED** [review]

**Review by Popov72 — APPROVED** [review]

**Review by sebavan — COMMENTED** [review]
Would be great to add a breaking change due to the new initialization way of recast ???

**Comment by CedricGuillemet** [comment]
@sebavan yes, preparing a new PR for that.

**Review by sebavan — APPROVED** [review]

**Pull request merged**